hphp/util/hash-crc-arm.S

```diff
@@ -0,0 +1,189 @@
+#include "hphp/util/etch-helpers.h"
+#include "hphp/util/hphp-config.h"
+
+#if defined(ENABLE_AARCH64_CRC) && !defined(NO_HWCRC) && \
+    !defined(__CYGWIN__) && !defined(__MINGW__) && !defined(_MSC_VER)
+
+.file   "hphp/util/hash-crc-arm.S"
+
+#define HASH_FUNC_NAME _ZN4HPHP20hash_string_i_unsafeEPKcj
+
+ETCH_SECTION(HASH_FUNC_NAME)
+.globl    HASH_FUNC_NAME
+ETCH_TYPE(HASH_FUNC_NAME, @function)
+ETCH_ALIGN16
+ETCH_NAME(HASH_FUNC_NAME):
+        CFI(startproc)
+        negs    w1, w1
+        mov     w9, #0xffffffff
+        beq     ETCH_LABEL(iend)
+        mov     w3, w1
+        mov     x4, #0xdfdfdfdfdfdfdfdf
+        b       ETCH_LABEL(iheader)
+
+ETCH_ALIGN16
```

Comment:
Hmm, the Cortex-A72 and A73 appear to have fetchers which are 3 instructions wide [A72](http://www.anandtech.com/show/10347/arm-cortex-a73-artemis-unveiled/2) [A73](http://techreport.com/review/28189/inside-arm-cortex-a72-microarchitecture).